metals/src/main/scala/scala/meta/internal/metals/MetalsLspService.scala

```diff
@@ -1214,6 +1217,13 @@ class MetalsLspService(
   def didChangeWatchedFiles(
       events: List[FileEvent]
   ): Future[Unit] = {
+    def isNewBuildTool(path: AbsolutePath): Boolean = {
+      path.isInBspDirectory(folder) &&
+      path.extension == "json" &&
+      !tables.buildServers
+        .selectedServer()
+        .contains(path.filename.stripSuffix(".json"))
```

Comment:
I think it works good? If the currently chosen server is the same as the one that appeared in the `.bsp`, we don't want to call `quickConnectToBuildServer`, because it's most likely already been called in other place, but if some other build server appears we want to notify the user?

Reply:
So maybe if the newly added build tool is different from the one in `tables.buildServers`, we can clean this table and then call `quickConnectToBuildServer`?

Reply:
We can wait for someone invoking `Switch build server command` in that case I think. If they are already using a bsp server. This will be a rare enough scenario.